app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+
+  def update_post_counter(id)
+    new_counter = Post.where(author_id: id).count
+    User.where(id: id).update(posts_counter: new_counter)
+  end
```

Comment:
- [ ] kindly work on all the methods that are performing increments, passing a parameter to a method that will not be used else where becomes unecessarry. We can also make use of the active record callback methods like `after_save` or `after_update`, and hide our `update_posts_counter` method under private, something like so:
```
after_save: update_post_counter

private
def update_post_counter
    author.increment!(:posts_counter)
 end
```